### ISSUE TITLE
Wancho online keyboard Shift key not working on mobile

### ISSUE THREAD

**Issue by punchcutter** [keymanapp/keyboards#1345]
https://keymanweb.com/?_ga=2.203677015.615597826.1600810924-368246240.1600562164&fbclid=IwAR1BRY40CyUQVHyp_JSq_o1JUs0AKuTSw49gxSBXqWl7FUuq--7zb_DWgGk#nnp-wcho,Keyboard_wancho
The keyboard seems fine on desktop browsers, but the shift key doesn't seem to be working on Android or iPhone.

**Comment by LornaSIL** [comment]
I am able to use the shift key on my Android using the Keyman app.

**Comment by punchcutter** [comment]
It works fine for me with the separate Keyman app, but not from the online keymanweb.com keyboard.

**Comment by LornaSIL** [comment]
@mcdurdin This seems to be true for me too. It works in my Keyman app on Android. It works on my web browser in my PC, but the shift key doesn't do anything on my web browser (Chrome) in my Android phone. Is this a bug in the keyboard or in Keyman?

**Comment by darcywong00** [comment]
I can replicate the issue on web browser in the PC (mobile viewport).

Looking at the wancho.keyman-touch-layout, the <kbd>Shift</kbd> key doesn't set the next layer, so it stays on the default layer.
```
              {
                "id": "K_SHIFT",
                "text": "*Shift*",
                "width": "160",
                "sp": "1"
              },
```
I believe setting the next layer to "shift" (adding` "nextlayer": "shift"`) will solve this

Don't forget to fix the Shift key on the Ctrl layer too.

**Comment by mcdurdin** [comment]
@darcywong00 note that it seems it doesn't work in web but does in mobile apps? I think we need to get to the bottom of why it's behaving differently, too.

**Comment by darcywong00** [comment]
> I am able to use the shift key on my Android using the Keyman app.

@LornaSIL , so you're able to get to the shift layer in Keyman for Android 13.0?

On 14.0 alpha, I installed [Wancho 1.0](https://downloads.keyman.com/keyboards/wancho/1.0/wancho.kmp) but couldn't get to the shift layer.

**Comment by LornaSIL** [comment]
I'm on Android 8.0.0
Keyman 13.0.6218

**Comment by darcywong00** [comment]
Hmm, I'm stumped.
I've got Android 8.1
Keyman 13.6216 (close enough)
I installed **Naga (Wancho)** from the cloud, and wasn't able to get to the shift layer.
I uninstalled that, and installed the Wancho keyboard package, and still couldn't get to the shift layer.

**Comment by mcdurdin** [comment]
Is it possible, @LornaSIL, that you have a different version of the keyboard on your phone? I'm finding it hard to see how it could be behaving in the way you are experiencing, otherwise. @darcywong00's suggestion on how to resolve the problem seems right.

**Comment by LornaSIL** [comment]
Well...I don't know. Since I helped get Wancho online it is entirely possible that I installed it while testing. I just uninstalled the keyboard and then reinstalled it. The shift key still works...but it says it's the Wancho language (not Wancho Naga) so that indicates to me that it's installing from my phone, not from the cloud version, correct?

Anyway, I just looked at the source and Darcy is right. The code to go to the shift layer is not there.
I'll update the keyboard including removing the "ctrl" layer.

**Comment by mcdurdin** [comment]
> The shift key still works...but it says it's the Wancho language (not Wancho Naga) so that indicates to me that it's installing from my phone, not from the cloud version, correct?

I guess it's possible it's a cached file -- in which case you'd have to uninstall Keyman entirely to sort that out, I think. I was confused by the mix of platform reports initially but I am guessing your phone is the only one experiencing the 'good' behaviour, so I'm happy to leave this here with the fix in #1350.
